File: ssTools/kukijson/python/kukijson.py
import sys
import json


def formatData(data):
    namaArray = data["nama"]
    isiArray = data["isi"]

    if len(namaArray) != len(isiArray):
        return ""

    formattedString = ""

    for i in range(len(namaArray)):
        formattedString += f"{namaArray[i]}={isiArray[i]}"

        if i != len(namaArray) - 1:
            formattedString += ";"

    return formattedString


if len(sys.argv) != 3 or sys.argv[1] != "-f":
    print("Penggunaan: python program.py -f <nama_file.json>")
    sys.exit(1)

filename = sys.argv[2]


try:
    with open(filename) as file:
        jsonData = file.read()
except FileNotFoundError:
    print("File tidak ditemukan:", filename)
    sys.exit(1)
except IOError as e:
    print("Gagal membaca file:", filename)
    sys.exit(1)

try:
    data = json.loads(jsonData)
except json.JSONDecodeError as e:
    print("Gagal memparse JSON:", str(e))
    sys.exit(1)

print(formatData(data))

File: ssTools/kukijson/js/kukijson.js
const fs = require("fs");

const args = process.argv.slice(2);

if (args.length !== 2 || args[0] !== "-f") {
	console.log("Penggunaan: node program.js -f <nama_file.json>");
	process.exit(1);
}

const filename = args[1];

fs.readFile(filename, "utf8", (err, data) => {
	if (err) {
		console.error("Gagal membaca file:", err);
		process.exit(1);
	}

	let jsonData;
	try {
		jsonData = JSON.parse(data);
	} catch (err) {
		console.error("Gagal memparse JSON:", err);
		process.exit(1);
	}

	const formattedData = formatData(jsonData);
	console.log(formattedData);
});

function formatData(data) {
	const namaArray = data.nama;
	const isiArray = data.isi;

	if (namaArray.length !== isiArray.length) {
		return "";
	}

	let formattedString = "";

	for (let i = 0; i < namaArray.length; i++) {
		formattedString += `${namaArray[i]}=${isiArray[i]}`;

		if (i !== namaArray.length - 1) {
			formattedString += ";";
		}
	}

	return formattedString;
}
